pkg/appgw/cleanup.go

```diff
@@ -0,0 +1,136 @@
+// -------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// --------------------------------------------------------------------------------------------
+
+package appgw
+
+import (
+	"strings"
+
+	n "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-03-01/network"
+)
+
+// CleanUpUnusedDefaults removes the default backend and default http settings if they are not used by any ingress
+func (c *appGwConfigBuilder) CleanUpUnusedDefaults() {
+	if !c.isPoolUsed(DefaultBackendAddressPoolName) {
+		c.removePool(DefaultBackendAddressPoolName)
+	}
+
+	if !c.isBackendSettingsUsed(DefaultBackendHTTPSettingsName) {
+		c.removeBackendSettings(DefaultBackendHTTPSettingsName)
+	}
+
+	if !c.isProbeUsed(defaultProbeName(n.ApplicationGatewayProtocolHTTP)) {
+		c.removeProbe(defaultProbeName(n.ApplicationGatewayProtocolHTTP))
+	}
+
+	if !c.isProbeUsed(defaultProbeName(n.ApplicationGatewayProtocolHTTPS)) {
+		c.removeProbe(defaultProbeName(n.ApplicationGatewayProtocolHTTPS))
+	}
+}
+
+func (c *appGwConfigBuilder) isPoolUsed(name string) bool {
```

Comment:
nit: might be nicer to name this function `isDefaultPoolUsed` just to dissuade anyone from using this function with a pool name more likely to have a conflict on the `HasSuffix` call below.

same for the other `removeX`  funcs

Reply:
changed from sufix comparison to name comparison.